xwiki-rendering-test/src/main/java/org/xwiki/rendering/test/integration/AbstractRenderingTest.java

```diff
@@ -102,6 +103,12 @@ public AbstractRenderingTest(String input, String expected, String parserId, Str
 
     public void execute() throws Exception
     {
+        // Register the context component manager.
+        DefaultComponentDescriptor<Object> contextCM = new DefaultComponentDescriptor<>();
+        contextCM.setRoleType(ComponentManager.class);
+        contextCM.setRoleHint("context");
+        getComponentManager().registerComponent(contextCM, getComponentManager());
```

Comment:
@surli hmmm, I guess the main issue is that there's no concept of "context" CM in commons/rendering and thus there's no component implementation for it. It's a bit weird to have components in rendering relying on "context" CM since it doesn't exist at this level (the context CM will look in the current user scope, then current wiki scope, etc).

I guess it's acceptable if we consider that:
1) There's a notion of ExecutionContext in xwiki-commons
2) We consider the term "context" in a vague way and thus consider that it would be a way to get a CM depending on some context (whatever it is).
3) However we need a "context" CM implementation in xwiki-rendering (with a low priority so that it's overridden by the platform one) since otherwise it means that rendering code cannot be used independently of xwiki-platform and that would be bad since it's the main reason we have it in a separate repo.

Thus, to answer you, IMO we just need that implementation (which should just do nothing and return the main CM by default). Once you have that you don't need to set up anything specific for the tests.

Reply:
No, commons and rendering are supposed to use the cotext ComponentManager **Provider** which is taking care of checking if a context ComponentManager exist or fallback on the root ComponentManager otherwise.

Reply:
Thanks for the information, so indeed using a `Provider` instead of injecting directly the CM is working there.